oap-ape/ape-java/ape-spark/pom.xml

```diff
@@ -50,136 +51,204 @@
+                <jvmArg>-XX:ReservedCodeCacheSize=512m</jvmArg>
+              </jvmArgs>
+              <javacArgs>
+                <javacArg>-source</javacArg>
+                <javacArg>${java.version}</javacArg>
+                <javacArg>-target</javacArg>
+                <javacArg>${java.version}</javacArg>
+                <javacArg>-Xlint:all,-serial,-path</javacArg>
+              </javacArgs>
+            </configuration>
+          </plugin>
+          <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-checkstyle-plugin</artifactId>
+            <version>3.0.0</version>
+            <configuration>
+              <failOnViolation>true</failOnViolation>
+              <includeTestSourceDirectory>true</includeTestSourceDirectory>
+              <sourceDirectories>
+                <directory>src/main/java</directory>
+              </sourceDirectories>
+              <testSourceDirectories>
+                <directory>src/test/java</directory>
+              </testSourceDirectories>
+              <configLocation>${basedir}/../dev/checkstyle.xml</configLocation>
+              <outputFile>${basedir}/target/checkstyle-output.xml</outputFile>
+              <inputEncoding>${project.build.sourceEncoding}</inputEncoding>
+              <outputEncoding>${project.reporting.outputEncoding}</outputEncoding>
+            </configuration>
+            <executions>
+              <execution>
+                <phase>compile</phase>
+                <goals>
+                  <goal>check</goal>
+                </goals>
+              </execution>
+            </executions>
+          </plugin>
+          <plugin>
+            <groupId>org.scalastyle</groupId>
+            <artifactId>scalastyle-maven-plugin</artifactId>
+            <version>1.0.0</version>
+            <configuration>
+              <verbose>false</verbose>
+              <failOnViolation>true</failOnViolation>
+              <failOnWarning>false</failOnWarning>
+              <includeTestSourceDirectory>true</includeTestSourceDirectory>
+              <sourceDirectories>
+                <directory>src/main/scala</directory>
+              </sourceDirectories>
+              <testSourceDirectories>
+                <directory>src/test/scala</directory>
+              </testSourceDirectories>
+              <configLocation>${basedir}/../dev/scalastyle-config.xml</configLocation>
+              <outputFile>target/scalastyle-output.xml</outputFile>
+              <inputEncoding>UTF-8</inputEncoding>
+              <outputEncoding>UTF-8</outputEncoding>
+            </configuration>
+            <executions>
+              <execution>
+                <phase>compile</phase>
+                <goals>
+                  <goal>check</goal>
+                </goals>
+              </execution>
+            </executions>
+          </plugin>
+        </plugins>
+      </build>
+    </profile>
+
+    <profile>
+      <id>native</id>
```

Comment:
`native` should be built before `shading`. Is there anything that guarantees this?